File: android/Pets/app/src/main/java/com/pets/services/ServiceLocator.kt
package com.pets.services

import android.content.Context
import com.pets.services.Interfaces.DataService
import com.pets.services.Interfaces.FavoriteService
import com.pets.services.Interfaces.LoggingService

object ServiceLocator {
    val favoriteService: FavoriteService by lazy { FavoritesProvider() }
    val dataService: DataService by lazy { ServiceBuilder.buildService(DataService::class.java) }
    val localDataService: DataService by lazy { LocalDataProvider() }
    val logService: LoggingService by lazy { LogProvider() }
}

File: ios/Pets/Services/Sources/Services/ServiceLocator.swift
//
//  File.swift
//  
//
//  Created by Dylan Velez on 10/08/21.
//

import Foundation

public class ServiceLocator {
    public static var localService: DataService = LocalDataProvider()
    public static var httpService: DataService = HttpDataProvider()
    public static var favoriteService: FavoriteService = FavoritesProvider()
    public static var logService: LoggingService = LogProvider()
}
